schemas/stsci.edu/asdf/0.1.0/asdf-schema.yaml

```diff
@@ -0,0 +1,46 @@
+%YAML 1.1
+---
+$schema: "http://json-schema.org/draft-04/schema"
+id: "http://stsci.edu/schemas/asdf/0.1.0/asdf-schema"
+title:
+  ASDF schema
+description: |
+  A metaschema extending YAML Schema and JSON Schema to add support
+  for some ASDF-specific checks, related to nd-arrays.
```

Comment:
I agree--I still think the ASDF schema is nicer here.  I do worry about a slippery slope, where suddenly every schema author wants their own custom validators for their schema, and it becomes impossible for implementations to keep up.

But if we keep this at an "ASDF core" level (and in particular since this is for something as low-level and fundamental as array dimensions) I think it's fine.